### PR TITLE
feat: first commit for base system init test

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Unit tests for the `SystemInit` component in `mava/components/jax/building/base.py`.
## Why?
Expanding the testing of Mava Jax redesign.

### CODE REVIEW

File: tests/jax/components/building/system_init_test.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+    ],
+) -> None:
+    """Tests all system network sampling instantiation ValueErrors."""
+
+    (
+        fixed_network_system,
+        fixed_network_builder,
+    ) = fixed_agent_networks_with_incorrect_sampling_setup_type
+    with pytest.raises(ValueError):
+        assert fixed_network_system.on_building_init(fixed_network_builder) == ValueError  # type: ignore # noqa: E501
```
Comment (sash-a) [inline tests/jax/components/building/system_init_test.py:457]: I don't think you need the assert in here as well as the `with pytest.raises(ValueError):`, ditch the `assert ... == Error` and just use the pytest stuff

**Review by sash-a — CHANGES_REQUESTED** [review]
Other than the minor change, I feel like these are kind of superficial tests, you're checking that what you pass in is what you get out. Can you do 1 extra test where there's an edge case like maybe `unique_net_keys` aren't actually all unique, or `random_networks_net_keys_to_ids` assigns two agents to a single network?

**Review by arnupretorius — COMMENTED** [review]
Looks good, thanks @RuanJohn 🙂 Just see my one comment on the config.

File: tests/jax/components/building/system_init_test.py (hunk 1 above)
Comment (arnupretorius) [reply tests/jax/components/building/system_init_test.py:457]: Might also help with the type issues.

Comment (RuanJohn) [reply tests/jax/components/building/system_init_test.py:457]: Thank you for this catch. Updated :call_me_hand:

File: mava/components/jax/building/system_init.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def on_building_init(self, builder: SystemBuilder) -> None:
                 "Random sampling system init requires random_agent_networks sampling"
             )
 
-        if builder.store.shared_weights:
+        if self.config.shared_weights:
```
[the remark this answers is not in the document]
Comment (mmorris44) [reply mava/components/jax/building/system_init.py:175]: Then those configs will all have `shared_weights` as a parameter and the system will complain about duplicate entries. I think we should rather just set it in one component and then access it through the global config for the rest

[the remark this answers is not in the document]
Comment (RuanJohn) [reply mava/components/jax/building/system_init.py:175]: I think that is a good idea, added a `BaseSystemInitConfig` that all the other configs inherit from :call_me_hand:

[the remark this answers is not in the document]
Comment (RuanJohn) [reply mava/components/jax/building/system_init.py:175]: All tests in `tests/jax/components/building/system_init_test.py` still pass and I can run `examples/jax/debugging/simple_spread/feedforward/decentralised/run_mappo.py` with no issues @mmorris44

**Review by sash-a — APPROVED** [review]
My mistake about the extra tests, looks good!

File: mava/components/jax/building/system_init.py [hunk 3]
```diff
@@ -131,11 +135,10 @@ def config_class() -> Optional[Callable]:
 
 
 @dataclass
-class RandomSamplingSystemInitConfig:
+class RandomSamplingSystemInitConfig(BaseSystemInitConfig):
```
Comment (mmorris44) [inline mava/components/jax/building/system_init.py:138]: As discussed, this one should please not inherit from `BaseSystemInitConfig`, since it can never have shared weights.

Comment (RuanJohn) [reply mava/components/jax/building/system_init.py]: Updated :+1:

**Review by mmorris44 — APPROVED** [review]
Looks good, thanks @RuanJohn!
